src/EditorFeatures/CSharpTest/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -6405,6 +6405,31 @@ void M(int j, int[] array)
 }", options: PreferDiscard);
         }
 
+        [WorkItem(40336, "https://github.com/dotnet/roslyn/issues/40336")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task RedundantAssignment_ForStatementVariableDeclaration()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        for (int [|i|] = 0; ; ) 
```

Comment:
Can you please add a test for non-constant value, i.e.: `for (int [|i|] = GetValue(); ; )`

Reply:
At the moment the suggested fix is replacing `i` with a discard pattern so `for (int [|i|] = GetValue(); ; )` would turn into `for (int _ = GetValue(); ; )`. 

Should we fix the expression to `for (GetValue(); ; )` since `GetValue()` might have side effects or completely get rid of the call (which might lead to unexpected removal of code when fixing all in locations the user might not immediately expect)?

Reply:
Correction: After investigating a bit it seems the `for (int _ = GetValue(); ; )` [is intended](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs#L461-L463). I therefore just added two tests for the current functionality.

Reply:
> I therefore just added two tests for the current functionality.

Thanks, yes I presume your change should not have any effect on these cases, but worth adding the unit tests validating the current behavior for completeness.